### ISSUE TITLE
Package Harvest doesn't probe NuGetFallbackFolder

### ISSUE THREAD

**Issue by joperezr** [dotnet/arcade#1351]
Now that we use the machine's cache for restoring packages in corefx, restore won't download packages that are part of the CLI's NuGetFallbackFolder. This causes an issue with harvesting, since that logic will only probe into the local cache for getting the assets from stable packages, so if you try to harvest from a package that comes in the fallback folder, harvesting will fail. I believe this will actually break brand new machines today if they try to build corefx master. The reason is that today we harvest from some package versions that come with the CLI that we use to build. It happens to work on older machines because you might have had restored that package version earlier, before using the new CLI.

cc: @ericstj

**Comment by ericstj** [comment]
Yep, this broke when I moved us to use the built-in restore target that honors machine wide SDK.

The fix here would be to pass in NuGetPackageFolders and probe them, like we do here: https://github.com/dotnet/arcade/blob/995d5bc5507f25475368368ecd6fef4252e8232d/src/Microsoft.DotNet.Build.Tasks.Packaging/tests/Microsoft.DotNet.Build.Tasks.Packaging.Tests.csproj#L38-L56

**Comment by ericstj** [comment]
We need to make this an array and pass in NuGetPackagesFolder:
https://github.com/dotnet/arcade/blob/995d5bc5507f25475368368ecd6fef4252e8232d/src/Microsoft.DotNet.Build.Tasks.Packaging/src/HarvestPackage.cs#L35
